tests/foreman/api/test_ansible.py

```diff
@@ -301,9 +301,11 @@ def test_positive_read_facts_with_filter(
         host.organization = module_org
         host.location = module_location
         host.update(['organization', 'location'])
-        request.addfinalizer(
-            user.delete
-        )  # Adding a temporary workaround until the issue 'SAT-18656' is resolved.
+        if is_open('SAT-18656'):
+
+            @request.addfinalizer
+            def _finalize():
+                target_sat.cli.Host.disassociate({'name': rex_contenthost.hostname})
```

Comment:
How about deleting the host record here instead of disassociating it, could you try if it works?
```suggestion
                target_sat.cli.Host.disassociate({'name': host.name})
```

Reply:
I guess while unregistering the host in teardown, it deletes the host record. I do not want to duplicate it.

Reply:
yes, I know with `_delete_host_record` we delete it in teardown, but since you're disassociating host here, could you confirm if it still works?

Reply:
Deleting the host record works locally but again got into same FK violation. This works locally but not on CI so I think we should use the previous approach.

Reply:
Also, I feel this is right approach to disassociate the host and let teardown do the rest what it is supposed to do.